examples/connext_dds/network_capture/example1-tcp/c/example1_publisher.c

```diff
@@ -0,0 +1,285 @@
+     *
+     * A capture file will be created for each participant. The capture file
+     * will start with the prefix "publisher" and continue with a suffix
+     * dependent on the participant's GUID.
+     */
+    if (!NDDS_Utility_start_network_capture("publisher")) {
+        fprintf(stderr, "Error starting network capture for all participants\n");
+        return -1;
+    }
+
+    /*
+     * The example continues as the usual Connext DDS HelloWorld.
+     * Create entities, start communication, etc.
+     */
+    participant = DDS_DomainParticipantFactory_create_participant_with_profile(
+            DDS_TheParticipantFactory,
+            domainId,
+            "Library",
+            profile,
+            NULL, /* listener */
+            DDS_STATUS_MASK_NONE);
+    if (participant == NULL) {
+        fprintf(stderr, "create_participant error\n");
+        publisher_shutdown(participant);
+        return -1;
+    }
+
+    publisher = DDS_DomainParticipant_create_publisher(
+            participant,
+            &DDS_PUBLISHER_QOS_DEFAULT,
+            NULL, /* listener */
+            DDS_STATUS_MASK_NONE);
+    if (publisher == NULL) {
+        fprintf(stderr, "create_publisher error\n");
+        publisher_shutdown(participant);
+        return -1;
+    }
+
+    type_name = example1TypeSupport_get_type_name();
+    retcode = example1TypeSupport_register_type(
+            participant,
+            type_name);
+    if (retcode != DDS_RETCODE_OK) {
+        fprintf(stderr, "register_type error %d\n", retcode);
+        publisher_shutdown(participant);
+        return -1;
+    }
+
+    topic = DDS_DomainParticipant_create_topic(
+            participant,
+            "Example example1 using network capture",
+            type_name,
+            &DDS_TOPIC_QOS_DEFAULT,
+            NULL, /* listener */
+            DDS_STATUS_MASK_NONE);
+    if (topic == NULL) {
+        fprintf(stderr, "create_topic error\n");
+        publisher_shutdown(participant);
+        return -1;
+    }
+
+    writer = DDS_Publisher_create_datawriter(
+            publisher,
+            topic,
+            &DDS_DATAWRITER_QOS_DEFAULT,
+            NULL, /* listener */
+            DDS_STATUS_MASK_NONE);
+    if (writer == NULL) {
+        fprintf(stderr, "create_datawriter error\n");
+        publisher_shutdown(participant);
+        return -1;
+    }
+    example1_writer = example1DataWriter_narrow(writer);
+    if (example1_writer == NULL) {
+        fprintf(stderr, "DataWriter narrow error\n");
+        publisher_shutdown(participant);
+        return -1;
+    }
+
+    instance = example1TypeSupport_create_data_ex(DDS_BOOLEAN_TRUE);
+    if (instance == NULL) {
+        fprintf(stderr, "example1TypeSupport_create_data error\n");
+        publisher_shutdown(participant);
+        return -1;
+    }
+
+    for (count=0; (sample_count == 0) || (count < sample_count); ++count) {
+
+        printf("Writing example1, count %d\n", count);
+        fflush(stdout);
```

Comment:
It was part of the original example that I modified and it flushes the stdout buffer so that the contents are displayed.
However, I read that the buffer gets flushed every time that we have a new line. Therefore, I removed the fflush calls in all the PR, as they are not needed.
Thanks, Antonio.